packages/core/src/helpers/fee.ts

```diff
@@ -96,3 +104,41 @@ export async function injectCapacityAndPayFee(props: {
     after: createCapacitySnapshot(inputs, outputs),
   };
 }
+
+/**
+ * Specify an output to pay transaction fee.
+ */
+export async function payFeeByOutput(props: {
+  txSkeleton: helpers.TransactionSkeletonType;
+  outputIndex: number;
+  feeRate?: BIish;
+  config?: SporeConfig;
+}): Promise<helpers.TransactionSkeletonType> {
+  // Env
+  const config = props.config ?? getSporeConfig();
+  const feeRate = props.feeRate ?? (await getMinFeeRate(config.ckbNodeUrl));
+
+  // Get TransactionSkeleton
+  let txSkeleton = props.txSkeleton;
+
+  // Get target output cell
+  let outputs = txSkeleton.get('outputs');
+  const output = outputs.get(props.outputIndex);
+  if (!output) {
+    throw new Error(`Cannot pay fee by Transaction.outputs[${props.outputIndex}] because it does not exist`);
+  }
+
+  // Check can pay fee with free capacity
+  const minimalCellCapacity = helpers.minimalCellCapacityCompatible(output);
+  const outputCapacity = BI.from(output.cellOutput.capacity);
+  const freeCapacity = outputCapacity.sub(minimalCellCapacity);
+  const fee = calculateFeeByTransactionSkeleton(txSkeleton, feeRate);
+  if (freeCapacity.lt(fee)) {
+    throw new Error(`Cannot pay fee by Transaction.outputs[${props.outputIndex}] because it has no enough capacity`);
```

Comment:
For users, this is a newly constructed transaction. Should the prompt be 'input'

Reply:
This helper function is designed to "pay transction fee by a specified output", therefore the output cell must exist, and it must have enough free capacity to pay fee.